### ISSUE TITLE
[Bug] Truyenqq source error "Too many requests. Try again later"

### ISSUE THREAD

**Issue by FiorenMas** [KotatsuApp/kotatsu-parsers#1066]
### Source information

Truyenqq

### Steps to reproduce

Step 1: Open source Truyenqq, open any manga in homepage.
Step 2: Error appear, screenshot:
![image](https://github.com/user-attachments/assets/e7e6c23c-10a6-4175-a085-805c52e35325)
If I continue reloading the manga page 4-6 times, the error disappears, and I can load the chapter list
Step 3: After reload 4-6 times chapter list, and then click one chapter to reading error again.
![image](https://github.com/user-attachments/assets/035695bb-61eb-4480-9d54-67f65541edb7)
This issue still persists when downloading manga.
![image](https://github.com/user-attachments/assets/e0604761-3fa5-4605-b379-041defc2dcb1)
I believe the problem is that Cloudflare is detecting something. When I open the manga within the webview app, I encounter this error:
![image](https://github.com/user-attachments/assets/f6939eba-9f59-4af8-92f8-9bdd9aefca2f)

### Kotatsu version

7.5.2

### Android version

Android 11

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.

**Comment by dragonx943** [comment]
> ### Source information
> Truyenqq
> 
> ### Steps to reproduce
> Step 1: Open source Truyenqq, open any manga in homepage. Step 2: Error appear, screenshot: ![image](https://private-user-images.githubusercontent.com/102145692/367672634-e7e6c23c-10a6-4175-a085-805c52e35325.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjcyNjM0LWU3ZTZjMjNjLTEwYTYtNDE3NS1hMDg1LTgwNWM1MmUzNTMyNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00OGExNzA3ODE2NjFlOTVmMDgxODJlMzcxYTdiZjIzY2U5YjdhZDI0YjVkMjdhYTcxY2U5ZDFlNjA1OGIzMTIyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4cNz0kTWVAKqrI5FF02Kes74hJddcai6x21735YOWuU) If I continue reloading the manga page 4-6 times, the error disappears, and I can load the chapter list Step 3: After reload 4-6 times chapter list, and then click one chapter to reading error again. ![image](https://private-user-images.githubusercontent.com/102145692/367673195-035695bb-61eb-4480-9d54-67f65541edb7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjczMTk1LTAzNTY5NWJiLTYxZWItNDQ4MC05ZDU0LTY3ZjY1NTQxZWRiNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lYjExYWE3M2IwZWRhN2QwZmY2M2Q0YWQyMzQzZTVmMjcxODBhMGFjMjg3ZTY2Nzk3MWJlNDhmZmI1ZDZiZDA4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jS60flVujb-TT1KwFir9rOtUREgGat9kX7RYnirSk0Y) This issue still persists when downloading manga. ![image](https://private-user-images.githubusercontent.com/102145692/367673026-e0604761-3fa5-4605-b379-041defc2dcb1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjczMDI2LWUwNjA0NzYxLTNmYTUtNDYwNS1iMzc5LTA0MWRlZmMyZGNiMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NWExZjQyY2EwODVhZTIyZDA4Yzg0OWFhZGNjYTU1MWM0NTVmZGIwZjFiMDg5YmNlZTU3OTY1ZWMyYTEyNjM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Jbldz_CpIvRC4RIuBVnwgIrxIESE_yGC6SFE4sEZ3Zw) I believe the problem is that Cloudflare is detecting something. When I open the manga within the webview app, I encounter this error: ![image](https://private-user-images.githubusercontent.com/102145692/367673425-f6939eba-9f59-4af8-92f8-9bdd9aefca2f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjczNDI1LWY2OTM5ZWJhLTlmNTktNGFmOC05MmY4LTliZGQ5YWVmY2EyZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MGU1YzE2N2Y1MjIxYzk4NDgyNDAxNDM1ZjFjYTZlNjM4NzE1N2EyYWJkZGFiMDdkNzc4ZjVlYTBjOGViMmE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.k9K7e_nsLOPE92IFLoUvsi72Q0ye44ZEgX-HbGC9pRw)
> 
> ### Kotatsu version
> 7.5.2
> 
> ### Android version
> Android 11
> 
> ### Other details
> _No response_
> 
> ### Acknowledgements
> * [x]  I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.

blocked -> cant access source -> cant request data
u can use 1.1.1.1 or VPN to read with this source 👍

**Comment by FiorenMas** [comment]
> > ### Source information
> > Truyenqq
> > ### Steps to reproduce
> > Step 1: Open source Truyenqq, open any manga in homepage. Step 2: Error appear, screenshot: ![image](https://private-user-images.githubusercontent.com/102145692/367672634-e7e6c23c-10a6-4175-a085-805c52e35325.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjcyNjM0LWU3ZTZjMjNjLTEwYTYtNDE3NS1hMDg1LTgwNWM1MmUzNTMyNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00OGExNzA3ODE2NjFlOTVmMDgxODJlMzcxYTdiZjIzY2U5YjdhZDI0YjVkMjdhYTcxY2U5ZDFlNjA1OGIzMTIyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4cNz0kTWVAKqrI5FF02Kes74hJddcai6x21735YOWuU) If I continue reloading the manga page 4-6 times, the error disappears, and I can load the chapter list Step 3: After reload 4-6 times chapter list, and then click one chapter to reading error again. ![image](https://private-user-images.githubusercontent.com/102145692/367673195-035695bb-61eb-4480-9d54-67f65541edb7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjczMTk1LTAzNTY5NWJiLTYxZWItNDQ4MC05ZDU0LTY3ZjY1NTQxZWRiNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lYjExYWE3M2IwZWRhN2QwZmY2M2Q0YWQyMzQzZTVmMjcxODBhMGFjMjg3ZTY2Nzk3MWJlNDhmZmI1ZDZiZDA4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jS60flVujb-TT1KwFir9rOtUREgGat9kX7RYnirSk0Y) This issue still persists when downloading manga. ![image](https://private-user-images.githubusercontent.com/102145692/367673026-e0604761-3fa5-4605-b379-041defc2dcb1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjczMDI2LWUwNjA0NzYxLTNmYTUtNDYwNS1iMzc5LTA0MWRlZmMyZGNiMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NWExZjQyY2EwODVhZTIyZDA4Yzg0OWFhZGNjYTU1MWM0NTVmZGIwZjFiMDg5YmNlZTU3OTY1ZWMyYTEyNjM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Jbldz_CpIvRC4RIuBVnwgIrxIESE_yGC6SFE4sEZ3Zw) I believe the problem is that Cloudflare is detecting something. When I open the manga within the webview app, I encounter this error: ![image](https://private-user-images.githubusercontent.com/102145692/367673425-f6939eba-9f59-4af8-92f8-9bdd9aefca2f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjczNDI1LWY2OTM5ZWJhLTlmNTktNGFmOC05MmY4LTliZGQ5YWVmY2EyZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MGU1YzE2N2Y1MjIxYzk4NDgyNDAxNDM1ZjFjYTZlNjM4NzE1N2EyYWJkZGFiMDdkNzc4ZjVlYTBjOGViMmE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.k9K7e_nsLOPE92IFLoUvsi72Q0ye44ZEgX-HbGC9pRw)
> > ### Kotatsu version
> > 7.5.2
> > ### Android version
> > Android 11
> > ### Other details
> > _No response_
> > ### Acknowledgements
> > 
> > * [x]   I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
> 
> blocked -> cant access source -> cant request data u can use 1.1.1.1 or VPN to read with this source 👍

source not blocked, can access normal in browser like firefox or google chrome. Only error when use in-app browser in kotatsu. That is my mean, I think cloudflare detected kotatsu not normal browser so they blocked it.

**Comment by dragonx943** [comment]
> > > ### Source information
> > > Truyenqq
> > > ### Steps to reproduce
> > > Step 1: Open source Truyenqq, open any manga in homepage. Step 2: Error appear, screenshot: ![image](https://private-user-images.githubusercontent.com/102145692/367672634-e7e6c23c-10a6-4175-a085-805c52e35325.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjcyNjM0LWU3ZTZjMjNjLTEwYTYtNDE3NS1hMDg1LTgwNWM1MmUzNTMyNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00OGExNzA3ODE2NjFlOTVmMDgxODJlMzcxYTdiZjIzY2U5YjdhZDI0YjVkMjdhYTcxY2U5ZDFlNjA1OGIzMTIyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4cNz0kTWVAKqrI5FF02Kes74hJddcai6x21735YOWuU) If I continue reloading the manga page 4-6 times, the error disappears, and I can load the chapter list Step 3: After reload 4-6 times chapter list, and then click one chapter to reading error again. ![image](https://private-user-images.githubusercontent.com/102145692/367673195-035695bb-61eb-4480-9d54-67f65541edb7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjczMTk1LTAzNTY5NWJiLTYxZWItNDQ4MC05ZDU0LTY3ZjY1NTQxZWRiNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lYjExYWE3M2IwZWRhN2QwZmY2M2Q0YWQyMzQzZTVmMjcxODBhMGFjMjg3ZTY2Nzk3MWJlNDhmZmI1ZDZiZDA4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jS60flVujb-TT1KwFir9rOtUREgGat9kX7RYnirSk0Y) This issue still persists when downloading manga. ![image](https://private-user-images.githubusercontent.com/102145692/367673026-e0604761-3fa5-4605-b379-041defc2dcb1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjczMDI2LWUwNjA0NzYxLTNmYTUtNDYwNS1iMzc5LTA0MWRlZmMyZGNiMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NWExZjQyY2EwODVhZTIyZDA4Yzg0OWFhZGNjYTU1MWM0NTVmZGIwZjFiMDg5YmNlZTU3OTY1ZWMyYTEyNjM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Jbldz_CpIvRC4RIuBVnwgIrxIESE_yGC6SFE4sEZ3Zw) I believe the problem is that Cloudflare is detecting something. When I open the manga within the webview app, I encounter this error: ![image](https://private-user-images.githubusercontent.com/102145692/367673425-f6939eba-9f59-4af8-92f8-9bdd9aefca2f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjczNDI1LWY2OTM5ZWJhLTlmNTktNGFmOC05MmY4LTliZGQ5YWVmY2EyZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MGU1YzE2N2Y1MjIxYzk4NDgyNDAxNDM1ZjFjYTZlNjM4NzE1N2EyYWJkZGFiMDdkNzc4ZjVlYTBjOGViMmE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.k9K7e_nsLOPE92IFLoUvsi72Q0ye44ZEgX-HbGC9pRw)
> > > ### Kotatsu version
> > > 7.5.2
> > > ### Android version
> > > Android 11
> > > ### Other details
> > > _No response_
> > > ### Acknowledgements
> > > 
> > > * [x]    I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
> > 
> > 
> > blocked -> cant access source -> cant request data u can use 1.1.1.1 or VPN to read with this source 👍
> 
> source not blocked, can access normal in browser like firefox or google chrome. Only error when use in-app browser in kotatsu. That is my mean, I think cloudflare detected kotatsu not normal browser so they blocked it.

try to change user-agent, clear cache data / cookies or use VPN / CF (still works with me)

**Comment by FiorenMas** [comment]
> > > > ### Source information
> > > > Truyenqq
> > > > ### Steps to reproduce
> > > > Step 1: Open source Truyenqq, open any manga in homepage. Step 2: Error appear, screenshot: ![image](https://private-user-images.githubusercontent.com/102145692/367672634-e7e6c23c-10a6-4175-a085-805c52e35325.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjcyNjM0LWU3ZTZjMjNjLTEwYTYtNDE3NS1hMDg1LTgwNWM1MmUzNTMyNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00OGExNzA3ODE2NjFlOTVmMDgxODJlMzcxYTdiZjIzY2U5YjdhZDI0YjVkMjdhYTcxY2U5ZDFlNjA1OGIzMTIyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4cNz0kTWVAKqrI5FF02Kes74hJddcai6x21735YOWuU) If I continue reloading the manga page 4-6 times, the error disappears, and I can load the chapter list Step 3: After reload 4-6 times chapter list, and then click one chapter to reading error again. ![image](https://private-user-images.githubusercontent.com/102145692/367673195-035695bb-61eb-4480-9d54-67f65541edb7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjczMTk1LTAzNTY5NWJiLTYxZWItNDQ4MC05ZDU0LTY3ZjY1NTQxZWRiNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lYjExYWE3M2IwZWRhN2QwZmY2M2Q0YWQyMzQzZTVmMjcxODBhMGFjMjg3ZTY2Nzk3MWJlNDhmZmI1ZDZiZDA4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jS60flVujb-TT1KwFir9rOtUREgGat9kX7RYnirSk0Y) This issue still persists when downloading manga. ![image](https://private-user-images.githubusercontent.com/102145692/367673026-e0604761-3fa5-4605-b379-041defc2dcb1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjczMDI2LWUwNjA0NzYxLTNmYTUtNDYwNS1iMzc5LTA0MWRlZmMyZGNiMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NWExZjQyY2EwODVhZTIyZDA4Yzg0OWFhZGNjYTU1MWM0NTVmZGIwZjFiMDg5YmNlZTU3OTY1ZWMyYTEyNjM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Jbldz_CpIvRC4RIuBVnwgIrxIESE_yGC6SFE4sEZ3Zw) I believe the problem is that Cloudflare is detecting something. When I open the manga within the webview app, I encounter this error: ![image](https://private-user-images.githubusercontent.com/102145692/367673425-f6939eba-9f59-4af8-92f8-9bdd9aefca2f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjczNDI1LWY2OTM5ZWJhLTlmNTktNGFmOC05MmY4LTliZGQ5YWVmY2EyZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MGU1YzE2N2Y1MjIxYzk4NDgyNDAxNDM1ZjFjYTZlNjM4NzE1N2EyYWJkZGFiMDdkNzc4ZjVlYTBjOGViMmE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.k9K7e_nsLOPE92IFLoUvsi72Q0ye44ZEgX-HbGC9pRw)
> > > > ### Kotatsu version
> > > > 7.5.2
> > > > ### Android version
> > > > Android 11
> > > > ### Other details
> > > > _No response_
> > > > ### Acknowledgements
> > > > 
> > > > * [x]     I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
> > > 
> > > 
> > > blocked -> cant access source -> cant request data u can use 1.1.1.1 or VPN to read with this source 👍
> > 
> > 
> > source not blocked, can access normal in browser like firefox or google chrome. Only error when use in-app browser in kotatsu. That is my mean, I think cloudflare detected kotatsu not normal browser so they blocked it.
> 
> try to change user-agent, clear cache data / cookies or use VPN / CF (still works with me)

I captured a video about this issue. You can watch it here:
https://streamable.com/3a94lc
As you can see, I'm already using Cloudflare Warp and have cleared my cookies, but the issue persists.
If tried click chapter so many times about 5-6 times, can watch normaly so I think this not source blocked. And then I tried open in firefox not have any problem loading page.

**Comment by dragonx943** [comment]
> > > > > ### Source information
> > > > > Truyenqq
> > > > > ### Steps to reproduce
> > > > > Step 1: Open source Truyenqq, open any manga in homepage. Step 2: Error appear, screenshot: ![image](https://private-user-images.githubusercontent.com/102145692/367672634-e7e6c23c-10a6-4175-a085-805c52e35325.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjcyNjM0LWU3ZTZjMjNjLTEwYTYtNDE3NS1hMDg1LTgwNWM1MmUzNTMyNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00OGExNzA3ODE2NjFlOTVmMDgxODJlMzcxYTdiZjIzY2U5YjdhZDI0YjVkMjdhYTcxY2U5ZDFlNjA1OGIzMTIyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4cNz0kTWVAKqrI5FF02Kes74hJddcai6x21735YOWuU) If I continue reloading the manga page 4-6 times, the error disappears, and I can load the chapter list Step 3: After reload 4-6 times chapter list, and then click one chapter to reading error again. ![image](https://private-user-images.githubusercontent.com/102145692/367673195-035695bb-61eb-4480-9d54-67f65541edb7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjczMTk1LTAzNTY5NWJiLTYxZWItNDQ4MC05ZDU0LTY3ZjY1NTQxZWRiNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lYjExYWE3M2IwZWRhN2QwZmY2M2Q0YWQyMzQzZTVmMjcxODBhMGFjMjg3ZTY2Nzk3MWJlNDhmZmI1ZDZiZDA4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jS60flVujb-TT1KwFir9rOtUREgGat9kX7RYnirSk0Y) This issue still persists when downloading manga. ![image](https://private-user-images.githubusercontent.com/102145692/367673026-e0604761-3fa5-4605-b379-041defc2dcb1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjczMDI2LWUwNjA0NzYxLTNmYTUtNDYwNS1iMzc5LTA0MWRlZmMyZGNiMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NWExZjQyY2EwODVhZTIyZDA4Yzg0OWFhZGNjYTU1MWM0NTVmZGIwZjFiMDg5YmNlZTU3OTY1ZWMyYTEyNjM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Jbldz_CpIvRC4RIuBVnwgIrxIESE_yGC6SFE4sEZ3Zw) I believe the problem is that Cloudflare is detecting something. When I open the manga within the webview app, I encounter this error: ![image](https://private-user-images.githubusercontent.com/102145692/367673425-f6939eba-9f59-4af8-92f8-9bdd9aefca2f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1OTQ5NDIsIm5iZiI6MTcyNjU5NDY0MiwicGF0aCI6Ii8xMDIxNDU2OTIvMzY3NjczNDI1LWY2OTM5ZWJhLTlmNTktNGFmOC05MmY4LTliZGQ5YWVmY2EyZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QxNzM3MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MGU1YzE2N2Y1MjIxYzk4NDgyNDAxNDM1ZjFjYTZlNjM4NzE1N2EyYWJkZGFiMDdkNzc4ZjVlYTBjOGViMmE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.k9K7e_nsLOPE92IFLoUvsi72Q0ye44ZEgX-HbGC9pRw)
> > > > > ### Kotatsu version
> > > > > 7.5.2
> > > > > ### Android version
> > > > > Android 11
> > > > > ### Other details
> > > > > _No response_
> > > > > ### Acknowledgements
> > > > > 
> > > > > * [x]     I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
> > > > 
> > > > 
> > > > blocked -> cant access source -> cant request data u can use 1.1.1.1 or VPN to read with this source 👍
> > > 
> > > 
> > > source not blocked, can access normal in browser like firefox or google chrome. Only error when use in-app browser in kotatsu. That is my mean, I think cloudflare detected kotatsu not normal browser so they blocked it.
> > 
> > try to change user-agent, clear cache data / cookies or use VPN / CF (still works with me)
> 
> I captured a video about this issue. You can watch it here:
> https://streamable.com/3a94lc
> As you can see, I'm already using Cloudflare Warp and have cleared my cookies, but the issue persists.
> If tried click chapter so many times about 5-6 times, can watch normaly so I think this not source blocked. And then I tried open in firefox not have any problem loading page.

Too many requests is normal, just exit and retry until u can read and when reading another manga the above wont be repeated (maybe to avoid IP ban by Cloudflare due to request spam), the rest is the source is not the problem, the problem is whether your network is blocking / being blocked by that site or not
 i![Screenshot_20240918-101032.png](https://github.com/user-attachments/assets/ca155fbb-10a4-4ee0-8587-5e8241b7447c)

![Screenshot_20240918-101045.png](https://github.com/user-attachments/assets/f6ffca55-728b-4975-a713-e7924c46813f)

![Screenshot_20240918-101050.png](https://github.com/user-attachments/assets/52500094-0384-4def-9c62-12beb007a71a)
